src/test/regress/expected/multi_alter_table_add_constraints.out

```diff
@@ -488,3 +488,93 @@ SELECT "Constraint", "Definition" FROM table_checks WHERE relid='public.products
 
 \c - - - :master_port
 DROP TABLE products;
+SET citus.shard_count to 2;
+-- Test if the ALTER TABLE %s ADD %s PRIMARY KEY %s works
+CREATE SCHEMA sc1;
```

Comment:
There is a behavior mentioned in Postgres docs which makes me think that requires further investigation.

`If a constraint name is provided then the index will be renamed to match the constraint name. Otherwise the constraint will be named the same as the index.`

This means that, if you provide and arbitrary name, you'd change the index names on the workers. See the following:

```SQL
CREATE SCHEMA sc2;
CREATE TABLE sc2.alter_add_prim_key(x int, y int);
CREATE UNIQUE INDEX CONCURRENTLY alter_pk_idx ON sc2.alter_add_prim_key(x);
SELECT create_distributed_table('sc2.alter_add_prim_key', 'x');

ALTER TABLE sc2.alter_add_prim_key ADD CONSTRAINT onder PRIMARY KEY USING INDEX alter_pk_idx;

\d sc2.alter_add_prim_key 
          Table "sc2.alter_add_prim_key"
 Column |  Type   | Collation | Nullable | Default 
--------+---------+-----------+----------+---------
 x      | integer |           | not null | 
 y      | integer |           |          | 
Indexes:
    "onder" PRIMARY KEY, btree (x)

postgres=# \c - - - 9700
You are now connected to database "postgres" as user "onderkalaci" via socket in "/tmp" at port "9700".

\d sc2.alter_add_prim_key_102113
       Table "sc2.alter_add_prim_key_102113"
 Column |  Type   | Collation | Nullable | Default 
--------+---------+-----------+----------+---------
 x      | integer |           | not null | 
 y      | integer |           |          | 
Indexes:
    "onder_102113" PRIMARY KEY, btree (x)

```

I haven't hit any issues, but, could you ensure that we're good even if the index name changes? Please check Citus regression tests and see if there are any edge cases that could lead to problems since the index name has changed. (And add a test here as well)

My second note would be on `ALTER TABLE ... DROP CONSTRAINT`. Could you both manually test and add some tests here that we're all good with `ALTER TABLE sc2.alter_add_prim_key DROP CONSTRAINT alter_pk_idx;`  (i.e., make sure that we're also good with changed index names etc.)